nautobot/dcim/api/urls.py

```diff
@@ -81,5 +81,10 @@
 router.register("controllers", views.ControllerViewSet)
 router.register("controller-managed-device-groups", views.ControllerManagedDeviceGroupViewSet)
 
+# Modules
+router.register("modules", views.ModuleViewSet)
+router.register("module-types", views.ModuleTypeViewSet)
+router.register("module-bays", views.ModuleBayViewSet)
```

Comment:
No URL/viewset for ModuleBayTemplate?

Reply:
With the complexity of the models I'd like to focus on only the model, factory and model test code in this PR and save the Rest API for the next PR, hence the incomplete implementation here. I can go ahead and move this incomplete API code to another branch to make this review easier.